Source/Meadow.Foundation.Peripherals/Sensors.Motion.Lsm303agr/Driver/Lsm303agr.cs

```diff
@@ -0,0 +1,223 @@
+﻿using Meadow.Hardware;
+using Meadow.Units;
+using System;
+using System.Threading.Tasks;
+
+namespace Meadow.Foundation.Sensors.Accelerometers
+{
+    /// <summary>
+    /// Represents a LSM303AGR is a system-in-package (SiP) that combines a 3D linear acceleration sensor and a 3D magnetic sensor
+    /// </summary>
+    public partial class Lsm303agr :
+        ByteCommsSensorBase<(Acceleration3D? Acceleration3D, MagneticField3D? MagneticField3D)>
+    {
+        /// <summary>
+        /// Event raised when acceleration changes
+        /// </summary>
+        public event EventHandler<IChangeResult<Acceleration3D>> Acceleration3DUpdated = delegate { };
+
+        /// <summary>
+        /// Event raised when magnetic field changes
+        /// </summary>
+        public event EventHandler<IChangeResult<MagneticField3D>> MagneticField3DUpdated = delegate { };
+
+        /// <summary>
+        /// Current Acceleration 3D
+        /// </summary>
+        public Acceleration3D? Acceleration3D => Conditions.Acceleration3D;
+
+        /// <summary>
+        /// Current Magnetic Field 3D
+        /// </summary>
+        public MagneticField3D? MagneticField3D => Conditions.MagneticField3D;
+
+        readonly II2cPeripheral i2cPeripheralAccel;
+        readonly II2cPeripheral i2cPeripheralMag;
+
+        /// <summary>
+        /// Create a new Lsm303agr instance
+        /// </summary>
+        /// <param name="i2cBus">The I2C bus connected to the sensor</param>
+        public Lsm303agr(II2cBus i2cBus)
+        {
+            i2cPeripheralAccel = new I2cPeripheral(i2cBus, (byte)Addresses.AddressAccel_0x19);
+            i2cPeripheralMag = new I2cPeripheral(i2cBus, (byte)Addresses.AddressMag_0x69);
+
+            Initialize();
+        }
+
+        /// <summary>
+        /// Initializes the LSM303AGR sensor
+        /// </summary>
+        void Initialize()
+        {
+            i2cPeripheralAccel.WriteRegister(ACC_CTRL_REG1_A, 0x57);
```

Comment:
minor nit, but these should probably be defined constants instead of magic numbers